docs/Development.md

```diff
@@ -0,0 +1,175 @@
+    1. Do not create a new JIRA if creating a change to address an existing issue in JIRA; add to the existing discussion and work instead
+    2. Look for existing pull requests that are linked from the JIRA, to understand if someone is already working on the JIRA
+2. If the change is new, then it usually needs a new JIRA. However, trivial changes, where the what should change is virtually the same as the how it should change do not require a JIRA. Example: Fix spelling error in javadoc
+3. If required, create a new JIRA:
+    1. Provide a descriptive Title. “Problem in XXXManager” is not sufficient. “IoTDB failed to start on jdk11 because jdk11 does not support -XX:+PrintGCDetail” is good.
+    2. Write a detailed Description. For bug reports, this should ideally include a short reproduction of the problem. For new features, it may include a design document.
+    3. Set required fields:
+        1. Issue Type. Generally, Bug, Improvement and New Feature are the only types used in IoTDB.
+        2. Priority. Set to Major or below; higher priorities are generally reserved for committers to set. The main exception is correctness or data-loss issues, which can be flagged as Blockers. JIRA tends to unfortunately conflate “size” and “importance” in its Priority field values. Their meaning is roughly:
+            1. Blocker: pointless to release without this change as the release would be unusable to a large minority of users. Correctness and data loss issues should be considered Blockers.
+            2. Critical: a large minority of users are missing important functionality without this, and/or a workaround is difficult
+            3. Major: a small minority of users are missing important functionality without this, and there is a workaround
+            4. Minor: a niche use case is missing some support, but it does not affect usage or is easily worked around
+            5. Trivial: a nice-to-have change but unlikely to be any problem in practice otherwise
+        3. Affects Version. For Bugs, assign at least one version that is known to exhibit the problem or need the change
+        4. Label. Not widely used, except for the following:
+            * correctness: a correctness issue
+            * data-loss: a data loss issue
+            * release-notes: the change’s effects need mention in release notes. The JIRA or pull request should include detail suitable for inclusion in release notes – see “Docs Text” below.
+            * starter: small, simple change suitable for new contributors
+        5. Docs Text: For issues that require an entry in the release notes, this should contain the information that the release manager should include in Release Notes. This should include a short summary of what behavior is impacted, and detail on what behavior changed. It can be provisionally filled out when the JIRA is opened, but will likely need to be updated with final details when the issue is resolved.
```

Comment:
```suggestion
        5. Docs Text: For issues that require an entry in the release notes, this should contain the information that the release manager should include. Issues should include a short summary of what behavior is impacted, and detail on what behavior changed. It can be provisionally filled out when the JIRA is opened, but will likely need to be updated with final details when the issue is resolved.
```